### ISSUE TITLE
missing existing vulnerabilities in the dependency check report

### ISSUE THREAD

**Issue by githubhs17** [dependency-check/DependencyCheck#1389]
### Project 1:
When scanning a directory <PROJECT_1> which includes a package.json with entry "log4js": "^2.3.12" the scan finds the vulnerability of timespan and reports this:
**timespan:2.3.0`
File Path: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/package.json?timespan
Referenced In Project/Scope:null: transitive
  Evidence
  Identifiers
    npm: timespan:2.3.0  Confidence:Highest**

Everything OK (log4js uses timespan)

### Project 2:
When scanning a directory <PROJECT_2> which includes a package.json **without** entry "log4js", but with a directory node_modules/timespan the scan doesn't find the vulnerability of timespan.
That's strange for me. In the scan log, I see that the node module timespan is found by the scan,

2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/.npmignore
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/CHANGELOG.md
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/LICENSE
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/README.md
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/browser/TimeSpan-1.2.js
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/browser/TimeSpan-1.2.min.js
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/docs/docco.css
2018-07-19 15:57:49,238 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/docs/time-span.html
2018-07-19 15:57:49,239 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/lib/time-span.js
2018-07-19 15:57:49,239 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json
2018-07-19 15:57:49,239 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/test/date-parser-test.js
2018-07-19 15:57:49,239 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/test/helpers.js
2018-07-19 15:57:49,239 org.owasp.dependencycheck.App:356
DEBUG - Found file /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/test/time-span-test.js

**_then there is an entry "Skipping analysis"_**

2018-07-19 15:57:49,952 org.owasp.dependencycheck.analyzer.AbstractNpmAnalyzer:97
DEBUG - Skipping analysis of node module: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json

_and more log entries regarding to timespan:_

2018-07-19 15:57:51,696 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (File Name Analyzer)
. . .
2018-07-19 15:57:54,301 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Dependency Merging Analyzer)
. . .
2018-07-19 15:57:54,333 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Dependency Merging Analyzer)
. . .
2018-07-19 15:57:54,378 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Version Filter Analyzer)
. . .
2018-07-19 15:57:54,408 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Version Filter Analyzer)
. . .
2018-07-19 15:57:54,631 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Hint Analyzer)
. . .
2018-07-19 15:57:54,667 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Hint Analyzer)
. . .
2018-07-19 15:57:57,050 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (CPE Analyzer)
. . .
2018-07-19 15:57:57,096 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (CPE Analyzer)
. . .
2018-07-19 15:57:57,152 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (False Positive Analyzer)
. . .
2018-07-19 15:57:57,173 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (False Positive Analyzer)
. . .
2018-07-19 15:57:57,209 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (NVD CVE Analyzer)
. . .
2018-07-19 15:57:57,231 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (NVD CVE Analyzer)
. . .
**_and another entry "Skipping analysis"_**

2018-07-19 15:57:57,342 org.owasp.dependencycheck.analyzer.AbstractNpmAnalyzer:97
DEBUG - Skipping analysis of node module: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json
. . .
2018-07-19 15:57:57,342 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Node Security Platform Analyzer)
. . .
**_and a third entry "Skipping analysis"_**

2018-07-19 15:57:57,342 org.owasp.dependencycheck.analyzer.AbstractNpmAnalyzer:97
DEBUG - Skipping analysis of node module: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json
. . .
**_more "Skipping analysis"_**

2018-07-19 15:57:57,420 org.owasp.dependencycheck.analyzer.AbstractNpmAnalyzer:97
DEBUG - Skipping analysis of node module: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json
. . .
2018-07-19 15:57:57,420 org.owasp.dependencycheck.AnalysisTask:86
DEBUG - Begin Analysis of '/home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json' (Node Security Platform Analyzer)
. . .
**_"Skipping analysis" again_**

2018-07-19 15:57:57,420 org.owasp.dependencycheck.analyzer.AbstractNpmAnalyzer:97
DEBUG - Skipping analysis of node module: /home/jenkins-master/workspace/build-pipeline-ocscli.scan/node_modules/timespan/package.json
. . .

And as said, in the dependency check report there is no vulnerability for timespan.

**This also happens to other vulnerable node modules, like hoek, tunnel-agent, ... They are not reported.**

**Comment by kilodalton** [comment]
`npm audit` currently provides more useful and actionable information for JS dependencies. 

The nodesecurity.io endpoint this code appears to use will not be available after 9/30/2018.

https://blog.npmjs.org/post/175511531085/the-node-security-platform-service-is-shutting

**Comment by jeremylong** [comment]
@githubhs17 - does project 2 actually use timespan?  Or is the code just in the node_modules directory and it is not actually used?

**Comment by githubhs17** [comment]
@kilodalton For npm audit you need npm@6. We have npm 5.6 on our prod system
@jeremylong timespan is actually used. We have an own node module log4js-client (not an official node module) which uses the official node module log4js. And the official node module log4js is using timespan.

The npm tree of our service:

```
08:57:36 + npm ls
08:57:42 prod-delivery-render@0.0.6-hstest-SNAPSHOT.6 /jenkins-ws/prod-deli....UEFA/service
08:57:42 ├── @types/bluebird@3.5.20
...
08:57:42 ├─┬ log4js-client@4.1.4                        <-- our own node module
08:57:42 │ ├── bluebird@3.5.1 deduped
08:57:42 │ ├── dx-request-context@3.0.8 deduped
08:57:42 │ ├─┬ log4js@2.10.0
08:57:42 │ │ ├─┬ amqplib@0.5.2
...
08:57:42 │ │ ├─┬ loggly@1.1.1
08:57:42 │ │ │ ├── json-stringify-safe@5.0.1 deduped
08:57:42 │ │ │ ├─┬ request@2.75.0
...
08:57:42 │ │ │ └── timespan@2.3.0
```

**Comment by jeremylong** [comment]
Changes have been made to the npm analyzer that may resolve this issue. We will be release 5.2.3 within the next 2 weeks.

**Comment by jeremylong** [comment]
This issue is believed to be resolved with 5.2.3. Please open a new issue if the problem persists.
